anxcloud/resource_object_storage_tenant.go

```diff
@@ -0,0 +1,217 @@
+package anxcloud
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+
+	"go.anx.io/go-anxcloud/pkg/api"
+	"go.anx.io/go-anxcloud/pkg/apis/common"
+	objectstoragev2 "go.anx.io/go-anxcloud/pkg/apis/objectstorage/v2"
+)
+
+func resourceObjectStorageTenant() *schema.Resource {
+	return &schema.Resource{
+		Description:   "This resource allows you to create and manage Object Storage tenants.",
+		CreateContext: resourceObjectStorageTenantCreate,
+		ReadContext:   resourceObjectStorageTenantRead,
+		UpdateContext: resourceObjectStorageTenantUpdate,
+		DeleteContext: resourceObjectStorageTenantDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(2 * time.Minute),
+			Read:   schema.DefaultTimeout(1 * time.Minute),
+			Update: schema.DefaultTimeout(2 * time.Minute),
+			Delete: schema.DefaultTimeout(2 * time.Minute),
+		},
+		Schema: schemaObjectStorageTenant(),
+	}
+}
+
+func schemaObjectStorageTenant() map[string]*schema.Schema {
+	return mergeSchemas(
+		schemaObjectStorageCommon(),
+		schemaObjectStorageState(),
+		map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "The name of the tenant.",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Description of the tenant (can be empty).",
+			},
+			"user_name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the tenant user to be used for API login.",
+			},
+			"password": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Sensitive:   true,
+				Description: "Password of the tenant user to be used for API login.",
+			},
+			"quota": {
+				Type:         schema.TypeFloat,
+				Required:     true,
+				ValidateFunc: validation.FloatAtLeast(1),
+				Description:  "Maximum number of bytes allowed for objects within buckets (must be > 0).",
+			},
+			"usage": {
+				Type:        schema.TypeFloat,
+				Computed:    true,
+				Description: "Current number of bytes used by the user within buckets.",
+			},
```

Comment:
I think it'd be beneficial for the self-documenting nature of this schema to add a `_in_bytes` suffix to this.

```suggestion
			"quota_in_bytes": {
				Type:         schema.TypeFloat,
				Required:     true,
				ValidateFunc: validation.FloatAtLeast(1),
				Description:  "Maximum number of bytes allowed for objects within buckets (must be > 0).",
			},
			"usage_in_bytes": {
				Type:        schema.TypeFloat,
				Computed:    true,
				Description: "Current number of bytes used by the user within buckets.",
			},
```